repo/manifest/committed_manifest_manager.go

```diff
@@ -98,31 +107,75 @@ func (m *committedManifestManager) commitEntries(ctx context.Context, entries ma
 // the lock via commitEntries()) and to compact existing committed entries during compaction
 // where the lock is already being held.
 // +checklocks:m.cmmu
-func (m *committedManifestManager) writeEntriesLocked(ctx context.Context, entries map[ID]*manifestEntry) (map[content.ID]bool, error) {
+func (m *committedManifestManager) writeEntriesLocked(ctx context.Context, entries map[ID]*inMemManifestEntry) (map[content.ID]bool, error) {
 	if len(entries) == 0 {
 		return nil, nil
 	}
 
-	man := manifest{}
+	switch m.formatVersion {
+	case 0:
+		res, err := m.writeEntriesLockedV0(ctx, entries)
+		if err != nil {
+			return nil, errors.Wrap(err, "writing manifests in v0 format")
+		}
 
-	for _, e := range entries {
-		man.Entries = append(man.Entries, e)
+		return res, nil
+
+	case 1:
+		res, err := m.writeEntriesLockedV1(ctx, entries)
+		if err != nil {
+			return nil, errors.Wrap(err, "writing manifests in v1 format")
+		}
+
+		return res, nil
 	}
 
-	var buf gather.WriteBuffer
-	defer buf.Close()
+	return nil, errors.Errorf("unsupported format version: %d", m.formatVersion)
+}
 
-	gz := gzip.NewWriter(&buf)
-	mustSucceed(json.NewEncoder(gz).Encode(man))
+// +checklocks:m.cmmu
+func (m *committedManifestManager) writeContentLocked(
+	ctx context.Context,
+	prefix content.IDPrefix,
+	data interface{},
+	buf *gather.WriteBuffer,
+) (content.ID, error) {
+	gz := gzip.NewWriter(buf)
+	mustSucceed(json.NewEncoder(gz).Encode(data))
```

Comment:
now that I think about it, you will lose a lot of benefits of compression across manifests. Say you have 1MB manifest content with 10000 of similar manifests concatenated together (V0 format)- that might compress down to 150KB or so, but when you individually compress each of 10000 objects (100 bytes each), they will compress to 60 bytes each, so you'll end up with 600KB vs 150KB